uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -117,6 +117,14 @@ protected function update($params) {
       );
     }
 
+    if ($this->isTOILWithPastDates($params) && $this->isCancelledStatus($params)) {
+      if (!$this->canCanCancelTOILWithPastDates($params)) {
+        throw new RuntimeException(
```

Comment:
Just curious why the use of `RuntimeException`. I'd be happy to see us adopt a more structured approach to exceptions, I usually just use `\Exception` now, but would be good to have some agreement on best practice.

Reply:
I agree. We could discuss that later. I followed the convention for the other exceptions thrown in the file.